packages/devtools_app/lib/src/code_size/file_import_container.dart

```diff
@@ -0,0 +1,222 @@
+            });
+
+            if (widget.onFileSelected != null)
+              widget.onFileSelected(importedFile);
+          },
+          child: const MaterialIconLabel(Icons.file_upload, 'Import File'),
+        ),
+      ],
+    );
+  }
+
+  Column _buildAnalyzeButton(String title) {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        const SizedBox(height: defaultSpacing),
+        Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            RaisedButton(
+              onPressed: importedFile != null
+                  ? () => widget.onAction(importedFile)
+                  : null,
+              child: MaterialIconLabel(
+                Icons.highlight,
+                title,
+              ),
+            ),
+          ],
+        ),
+      ],
+    );
+  }
+}
+
+class DualFileImportContainer extends StatefulWidget {
+  const DualFileImportContainer({
+    @required this.actionText,
+    @required this.onAction,
+    Key key,
+  }) : super(key: key);
+
+  /// The title of the action button.
+  final String actionText;
+
+  final Function(String oldFilePath, String newFilePath) onAction;
+
+  @override
+  _DualFileImportContainerState createState() =>
+      _DualFileImportContainerState();
+}
+
+class _DualFileImportContainerState extends State<DualFileImportContainer> {
+  String pathToOldFile;
+  String pathToNewFile;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        Expanded(
+          child: FileImportContainer(
+            title: 'Old',
+            onFileSelected: onOldFileImport,
+            importNewFile: false,
+          ),
+        ),
+        const SizedBox(width: defaultSpacing),
+        Center(child: _buildAnalyzeButton('Analyze Diff')),
+        const SizedBox(width: defaultSpacing),
+        Expanded(
+          child: FileImportContainer(
+            title: 'New',
+            onFileSelected: onNewFileImport,
+          ),
+        ),
+      ],
+    );
+  }
+
+  void onOldFileImport(String importedPathToOldFile) {
+    setState(() {
+      pathToOldFile = importedPathToOldFile;
+    });
+  }
+
+  void onNewFileImport(String importedPathToNewFile) {
+    setState(() {
+      pathToNewFile = importedPathToNewFile;
+    });
+  }
+
+  Column _buildAnalyzeButton(String title) {
+    return Column(
+      mainAxisAlignment: MainAxisAlignment.center,
+      children: [
+        const SizedBox(height: defaultSpacing),
+        Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            RaisedButton(
+              onPressed: pathToOldFile != null && pathToNewFile != null
+                  ? () => widget.onAction(pathToOldFile, pathToNewFile)
+                  : null,
+              child: MaterialIconLabel(
+                Icons.highlight,
+                title,
+              ),
+            ),
+          ],
+        ),
+      ],
+    );
+  }
```

Comment:
this is duplicated. Make this a top level private 
method at the bottom of this file

Reply:
so _buildActionButton would take a VoidCallback param `onPressed` and then in both places where you call _buildActionButton, you would pass in what you currently have for `onPressed` there.

For example,
```
_buildAnalyzeButton(
  onPressed:  pathToOldFile != null && pathToNewFile != null
                  ? () => widget.onAction(pathToOldFile, pathToNewFile)
                  : null,
);
```